LSP/tests/credit_analysis_spec.rb

```diff
@@ -0,0 +1,32 @@
+require './app/credit_analysis'
+require './app/analyst'
+require './app/automatic_decision_analyst'
+
+RSpec.describe CreditAnalysis do
+  context 'when analysing an application' do
+    context 'and analyst is an automatic analyst' do
+      it 'automatically analyses the application' do
```

Comment:
eu imagino que esse seja o teste que não tá funcionando e *acho* que, além do desafio do `sample` que você mencionou, ele também não funcionaria por conta da verificação de classe que tem no `CreditAnalysis` (no _`if analyst.class == AutomaticDecisionAnalyst`_)...

Minha sugestão:
- Por enquanto, "xita" o teste (coloca um `xit` ao invés de `it`), assim o teste não vai rodar
- Pra depois, mudar a forma que o `sample` e a verificação de classe são usados (talvez extrair alguma coisa, abstrair algum atributo, ....)